Add docs to Kubernetes per-Resource RBAC

tigrato opened the pull request:
This PR adds documentation for the Kubernetes per-Resource RBAC and supported verbs introduced in Teleport 14.

tigrato committed:
Add docs to Kubernetes per-Resource RBAC

This PR adds documentation for the Kubernetes per-Resource RBAC and
supported verbs introduced in Teleport 14.

Signed-off-by: Tiago Silva <tiago.silva@goteleport.com>

ptgott commented on docs/pages/access-controls/access-requests/resource-requests.mdx:
```diff
@@ -133,8 +133,33 @@ To request access to these resources, run
+`windows_desktop`. Teleport also supports searching and requesting access to
+resources within Kubernetes clusters, such as:
+
+- `pod`: Grant acess to individual pods.
```
I think we should avoid repeating "Grant access to individual..." in each item. Actually, since each term lines up with a Kubernetes resource, I think it's okay to list each one without explanation.

I would add this list to a `Details` box so it doesn't take up as much room on this page.

tigrato replied:
236fbd2b919631466e288b1f3a408e18dce5b000

ptgott commented on docs/pages/access-controls/access-requests/resource-requests.mdx:
````diff
@@ -355,21 +380,21 @@ label:
 ```
 
-#### `pod`
+#### Kubernetes Resources
````
```suggestion
#### Kubernetes resources
```
We use sentence case in headings

ptgott commented on docs/pages/access-controls/access-requests/resource-requests.mdx:
````diff
@@ -396,29 +421,35 @@ spec:
-$ tsh request create <Var name="pod-id" />
+$ tsh request create <Var name="resource-id" />
 ```
+###### Namespaced-scoped Resources
````
Now that we're getting six heading levels deep, I think we should move the whole "Restrict the resources a user can request access to" H3 into its own reference guide.

Otherwise, the H6s don't render very nicely:

<img width="968" alt="image" src="https://github.com/gravitational/teleport/assets/25250668/f97dcc14-cba3-486f-bf62-36735a6ae570">


We can either do this as part of this PR or a separate one.

tigrato replied:
I think it's better do it in a separate PR

ptgott replied:
What would you think about extracting the "Access Requests for Kubernetes resources" H5 into an H3 below "Next steps"? We could put it above "Allow reviewers to see the hostnames of SSH Nodes". 

This way, we wouldn't need to use an H6 for any headings. I'd really like to avoid having an H6, since these are pretty much unreadable.

ptgott commented on docs/pages/kubernetes-access/controls.mdx:
```diff
@@ -487,19 +487,75 @@ spec:
-  supports the value `pod`.
+- `kind`: The kind of resource to enable access to. Currently, Teleport supports
+the following kinds:
+  - `*`: Allow access to all resources.
```
I think we could change this to a table. This way, we could reduce repetition of "Controls access to" and "the `name` field...corresponds to...".

tigrato replied:
236fbd2b919631466e288b1f3a408e18dce5b000

ptgott commented on docs/pages/kubernetes-access/controls.mdx:
```diff
@@ -487,19 +487,75 @@ spec:
   `production` namespace.
 - `name`: The name of the pod to allow access to. In `kube-access`, this is
   the `webapp` pod.
+- `verbs`: The operations to allow on the resource. Currently, Teleport supports:
```
I would change this to a table as well

tigrato replied:
236fbd2b919631466e288b1f3a408e18dce5b000

tigrato committed:
initial review

tigrato committed:
fix lint

ptgott commented on docs/pages/access-controls/access-requests/resource-requests.mdx:
```diff
@@ -133,8 +133,51 @@ To request access to these resources, run
+`windows_desktop`. Teleport also supports searching and requesting access to
+resources within Kubernetes clusters.
+
+<Details title="List of Supported Kubernetes resources">
```
```suggestion
<Details title="List of supported Kubernetes resources">
```

ptgott commented on docs/pages/access-controls/access-requests/resource-requests.mdx:
```diff
@@ -133,8 +133,51 @@ To request access to these resources, run
+  title="Resource Access Requests to Kubernetes Namespaces"
+>
+
+Requesting access to a Kubernetes Namespace allows you to access all resources
```
```suggestion
Requesting access to a Kubernetes namespace allows you to access all resources
```

ptgott commented on docs/pages/access-controls/access-requests/resource-requests.mdx:
```diff
@@ -133,8 +133,51 @@ To request access to these resources, run
+>
+
+Requesting access to a Kubernetes Namespace allows you to access all resources
+in that namespace including the supported resources listed above. However, it
```
```suggestion
in that namespace, including the supported resources listed above. However, it
```

ptgott replied:
Does "the supported resources listed above" mean "node, kube_cluster, db, app, and windows_desktop"?

tigrato replied:
> the supported resources listed above

It was a copy typo.

ptgott commented on docs/pages/access-controls/access-requests/resource-requests.mdx:
````diff
@@ -396,29 +439,35 @@ spec:
-$ tsh request create <Var name="pod-id" />
+$ tsh request create <Var name="resource-id" />
 ```
+###### Namespaced-scoped Resources
````
```suggestion
###### Namespaced-scoped resources
```

ptgott commented on docs/pages/access-controls/access-requests/resource-requests.mdx:
````diff
@@ -429,11 +478,41 @@ the regular expression `/^nginx-[a-z0-9-]+$/`, run the following command:
 ```
 
-If a user has no access to a Kubernetes cluster, they can search the list of pods
+###### Cluster-scoped Resources
````
```suggestion
###### Cluster-scoped resources
```

ptgott reviewed (commented):
Second wave of comments. Left a suggestion for a quick way to restructure the guide to avoid H6-level headings. Everything else consists of minor suggestions.

tigrato committed:
Apply suggestions from code review

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

tigrato committed:
fix bad reference

ptgott commented:
I'm going to open a PR against this branch to suggest a structure of the `resource-requests` page that avoids needing an H6 heading!

ptgott commented:
Here's my PR: https://github.com/gravitational/teleport/pull/31637

ptgott committed:
Restructure the Resource Requests guide (#31637)

Move the subsection on requesting access to Kubernetes resources to an
H3. This way, we can avoid an unslightly H6 heading. Also, since this
section relates to requesting access, not restricting access, it's more
discoverable in a separate H3.
